trenches2.py

```diff
@@ -565,9 +565,231 @@ def __init__(self, trench_corners: Dict[int, TrenchCorner], trenches: List[Trenc
+                                   trenches: List[Trench]) -> Tuple[Dict[str, TrenchCorner], List[Trench], List[int]]:
+    """
+    Retrunes the:
+    new_trench_corners: The building Nodes and the new road sub-trench Nodes as TrenchCorner
+    new_trenches: The Building Trenches and the new road sub-trenches
+    trench_indexes_to_remove: A list if indexes of the "trenches" list that should be removed
+      since we have replaced them with sub-trenches
+    :param building_by_closest_trench: The Trench Info dictionary
+    :param trenches: The current list of trenches
+    :return: new_trench_corners, new_trenches, trench_indexes_to_remove
+    """
+    new_trenches: List[Trench] = list()
+    new_trench_corners: Dict[int, TrenchCorner] = dict()
+    node_id = 500000000
+    trench_indexes_to_remove = list()
+    # Loop over all the buildings that we found closest trenches for and create the building trenches
+    # and the new sub-trenches that should replace the current road trenches
+    for trench_index, building_trench_info in building_by_closest_trench.items():
+        # It's possible that this trench has no buildings that should be connected to it
+        if len(building_trench_info) > 0:
+            trench_indexes_to_remove.append(trench_index)
+            last_shortest_i = 0
+            trench = trenches[trench_index]
+            last_node = trench['u_for_edge']
+            # Since we are creating the sub-trenches from teh "u" node the the "v" node
+            # We have to order the buildings so we chain the sub-trenches correctly
+            # The Trench Info object order them selves based on how far they are from the "u" node of the road trench
+            building_trench_info.sort()
+            # Loop over the now sorted building infos and create a new building trench
+            # and the next sub-trench in teh chain
+            for closest_trench_info1 in building_trench_info:
+                # Create the new road sub-trench node
+                node_id += 1
+                new_v_node_id = node_id
+                new_v_node = TrenchCorner(x=closest_trench_info1.new_v_node["x"],
+                                          y=closest_trench_info1.new_v_node["y"],
+                                          trench_count=3,
+                                          u_node_id=closest_trench_info1.corner_u,
+                                          street_ids={},
+                                          node_for_adding=new_v_node_id,
+                                          )
+                if str(trench_index) not in new_trench_corners:
+                    new_trench_corners[str(trench_index)] = set()
+                new_trench_corners[str(trench_index)].add(new_v_node)
+
+                # Create the building Node
+                node_id += 1
+                building_node_id = node_id
+                building_node = TrenchCorner(x=closest_trench_info1.building_centroid_node["x"],
```

Comment:
Feature request: can we build a function so that we can get an extract of all the houses in the network to use for clustering?